includes/abstracts/abstract-wc-shipping-method.php

```diff
@@ -456,11 +462,13 @@ public function admin_options() {
 	public function get_option( $key, $empty_value = null ) {
 		// Instance options take priority over global options.
 		if ( $this->instance_id && array_key_exists( $key, $this->get_instance_form_fields() ) ) {
-			return $this->get_instance_option( $key, $empty_value );
+			$instance_option = apply_filters( 'woocommerce_shipping_' . $this->id . '_instance_option', $this->get_instance_option( $key, $empty_value ), $key, $empty_value, $this );
```

Comment:
This filter should move to the `get_instance_option` method. 

I'd recommend naming it:

```
woocommerce_shipping_METHODID_instance_option
```

And only passing in `$key` and `$this`.